### ISSUE TITLE
Why the use of masked_fill

### ISSUE THREAD

**Issue by tanmay2004** [huggingface/transformers#42789]
Hi, I was recently going through the implementation of the deepseek_v3 model and was confused by the use of `masked_fill` [here](https://github.com/huggingface/transformers/blob/main/src/transformers/models/deepseek_v3/modeling_deepseek_v3.py#L228). Can't this line be replaced by `scores_for_choice = router_logits_for_choice * score_mask` and do the same thing? It feels like the simpler way to apply the mask on the logits. I'm just curious about the decision to use `masked_fill` instead, and I was wondering if you would agree about replacing it with the simpler way I wrote above.

**Comment by nathon-lee** [comment]
hi，maybe：
masked_fill is preferred:

Ensures masked logits are sufficiently small (-inf) for strict zero probability in softmax
Avoids numerical precision issues
Makes the code intention more explicit

**Comment by Rocketknight1** [comment]
I expect both of those lines are equivalent, and neither is a big part of the runtime! My intuition is that `masked_fill` should be faster than multiplication because it doesn't need to actually do any arithmetic, but maybe the conversion to bool + negation on the mask array cancels that out.
